### ISSUE TITLE
Issue retrieving GPTVectorStoreIndex from storage

### ISSUE THREAD

**Issue by Donovan-Rucastle** [jerryjliu/llama_index#2237]
It seems as though there may be an issue with the V0.6.0 and loading indexes from storage. 

When generating the index for the first time and keeping it in memory, I am able to query the index to produce a summary and it works just as expected. In my instance, this involves looping through all the docs and creating an index for each to later be used in a graph. 

However, when saving the index to storage and reloading it at a later date, an error is produced anytime the index is queried. I have used a separate persist dir for each index which is passed through when loading them back in. 

Here is an example of my setup: 

```python
def load_all_indexes(index_files):
    indexes = []
    for index_file in index_files:
            storage_context = StorageContext.from_defaults(persist_dir=index_file)
            index = load_index_from_storage(storage_context = storage_context,  service_context=service_context)
            indexes.append(index)
    return indexes
```
```python
def create_summaries(indexes):
    summaries = []
    for index in indexes:
        retriever = index.as_retriever()
        query_engine = RetrieverQueryEngine.from_args(retriever, response_mode='tree_summarize')
        summary = query_engine.query("What is a summary of this document?")
        summaries.append(str(summary))
    return summaries
```
   
Here is an example of the error produced: 
    
 ```
/miniconda3/lib/python3.10/site-packages/llama_index/indices/vector_store/retrievers.py", line 90, in <listcomp>
    self._index.index_struct.nodes_dict[idx] for idx in query_result.ids
KeyError: 'cdb2c18c-b0d3-4af6-973d-0b0b69ff5ddd'
```

**Comment by Donovan-Rucastle** [comment]
Upon further inspection, it seems as though the issue lies in the way the index is generated. Previously I had generated an empty index, saved it to storage, and then inserted the document into it and saved it to storage again. 

Once I simplified the index creation process to just generate the index from the documents in the first instance, I was able to query the index after retrieving it from storage. Not sure if that is still a bug or just an incorrect way of initialising an index on my part.
